### PR TITLE
:sparkles: update CAPI to 1.8.1

### PR DESCRIPTION
**Pull request by lubedacht**
**What is the purpose of this pull request/Why do we need it?**

Updates CAPI version

**Description of changes:**

* sigs.k8s.io/cluster-api 1.7.4 => 1.8.1
* sigs.k8s.io/cluster-api/test 1.7.4 => 1.8.1
* k8s.io/api 0.29.7 => 0.30.3
* k8s.io/apimachinery 0.29.7 => 0.30.3
* k8s.io/client-go 0.29.7 => 0.30.3
* sigs.k8s.io/controller-runtime 0.17.5 => 0.18.5

**Special notes for your reviewer:**

**Checklist:**
- [x] Documentation updated
- [x] E2E Tests updated
- [x] Includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,147 +1,157 @@
 module github.com/ionos-cloud/cluster-api-provider-ionoscloud
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.2
```
Comment (mcbenjemaa) [inline go.mod:5]: it's better to set

```
go 1.22.2
```

and omit toolchain

File: test/e2e/helpers/finalizers.go [hunk 2]
```diff
@@ -21,32 +21,34 @@ package helpers
 
 import (
 	"fmt"
+	"k8s.io/apimachinery/pkg/types"
```
Comment (mcbenjemaa) [inline test/e2e/helpers/finalizers.go:24]: goimports

File: go.mod (hunk 1 above)
Comment (lubedacht) [reply go.mod:5]: Ah this was automatically added again. :angry:

**Review by mcbenjemaa — APPROVED** [review]
LGTM

Please reorder the imports.
```
builtin

upstream

module
```

**Comment by lubedacht** [comment]
The imports are automatically reordered by GCI. The linter would complain

**Review by mcbenjemaa — APPROVED** [review]

**Review by tinashe-mundangepfupfu — APPROVED** [review]

**Pull request merged**